server/src/modules/post/post.service.ts

```diff
@@ -128,14 +128,16 @@ export class PostService {
           attributes: {
             include: [
               [
-                Sequelize.fn('COUNT', Sequelize.col('topicId')),
-                'relatedTopics',
+                Sequelize.literal(
+                  `CASE WHEN TopicId = ${post.topicId} THEN 1 ELSE 0 END`,
```

Comment:
```suggestion
                  `CASE WHEN topicId = ${post.topicId} THEN 1 ELSE 0 END`,
```
Not very comfortable with this, partly because of a very small possibility of SQL injection happening by messing with the topicId value. Would it make more sense to adjust the `where` clause to take topic IDs into account?

Reply:
As discussed - the intent is to grab the most popular questions if we can't find posts with the same topicId. Related questions implementation to be made more robust once we have text search capabilities